src/pages/_app.tsx

```diff
@@ -27,6 +29,15 @@ if (process.env.NODE_ENV === 'development') {
 }
 
 function MyApp({ Component, pageProps }: AppProps) {
+  const [domLoaded, setDomLoaded] = useState(false);
+  useEffect(() => {
+    setDomLoaded(true);
+  }, []);
+
+  if (!domLoaded || typeof window === 'undefined') {
+    return null;
+  }
```

Comment:
제가 커밋으로 남겨놓긴 했는데 내용 첨부합니다!
- recoil effect 로 localstorage 에 user 값이 있냐 없냐에 따라 로그인 상태 여부를 판단하는데 해당 기능(localstorage 를 위한 recoil effect)을 추가할 때 Hydration failed 에러 발생
=> 유추한 결과 window undefined 를 판단하는데 이때 초기 렌더링 UI 와 달라 발생하는 거 같음
- 해당 URL 참고하여 수정 https://stackoverflow.com/questions/71706064/react-18-hydration-failed-because-the-initial-ui-does-not-match-what-was-render/71797054#71797054